src/mon/OSDMonitor.cc

```diff
@@ -11907,7 +11907,7 @@ bool OSDMonitor::_check_become_tier(
   const std::string &tier_pool_name = osdmap.get_pool_name(tier_pool_id);
   const std::string &base_pool_name = osdmap.get_pool_name(base_pool_id);
 
-  const FSMap &pending_fsmap = mon->mdsmon()->get_pending();
+  const FSMap &pending_fsmap = mon->mdsmon()->get_pending_fsmap();
```

Comment:
It doesn't look right to me either. Should I change it to `get_fsmap()`?

Reply:
The annoying part of this decision is that I'm not familiar enough with the MDSMonitor, and I don't know what decisions went into choosing using the pending map instead of the stable map. In any case, I do think relying on the stable map is the way to go, or then we must wait for the current pending to be committed before replying to the user. Personally, I'd go with the former as much as possible.

Reply:
@jecluis I would suggest we leave this as-is for now and open a separate issue to fix the systemic use of pending when the current epoch should be used. I'm not really certain either on the conversion and I don't want to delay #16608 when the current version of this PR resolves the issue found when thrashing max_mds: https://github.com/ceph/ceph/pull/16608#issuecomment-381352831

Reply:
@batrick alright, but lets fix these before mimic is released.

Reply:
http://tracker.ceph.com/issues/23768